libantworld/route_continuous.cc

```diff
@@ -7,28 +7,32 @@
 #include <limits>
 #include <tuple>
 
-// Standard C++ includes
+// Standard C includes
 #include <cassert>
-#include <cmath>
 
 // Libantworld includes
 #include "common.h"
 
-//----------------------------------------------------------------------------
-// Anonymous namespace
-//----------------------------------------------------------------------------
-namespace
+using namespace units::math;
+
+inline constexpr meter_t makeM(const GLfloat value)
 {
-float sqr(float x)
+    return units::make_unit<meter_t>(value);
+}
+
+template<class T1, class T2>
+auto
+distance2(const T1 &v1, const T2 &v2)
 {
-    return (x * x);
+    return hypot(v2[1] - v1[1], v2[0] - v1[0]);
 }
-//----------------------------------------------------------------------------
-float distanceSquared(float x1, float y1, float x2, float y2)
+
+template<class T>
+meter_t
+distance2(const T &v1, meter_t x2, meter_t y2)
```

Comment:
I feel the clarity of ``distanceSquared`` was worth the few characters - ``distance2`` makes me think it's version 2 somehow.